plumex/tests/test_regression_pipeline.py

```diff
@@ -123,3 +126,55 @@ def test_get_true_pred():
 
     np.testing.assert_array_almost_equal(expected, result_true)
     np.testing.assert_array_almost_equal(expected, result_pred)
+
+def test_do_sinusoid_regression():
+    expected = (1, 2, 3, 4)
+    a, w, g, b = expected
+
+    def sinusoid_func(t, r):
+        return a * np.sin(w * r - g * t) + b * r
+
+    axis = np.linspace(0, 1, 101)
+    tt, rr = np.meshgrid(axis, axis)
+
+    X = np.hstack((tt.reshape(-1, 1), rr.reshape(-1, 1)))
+    Y = sinusoid_func(tt, rr).reshape(-1)
+
+    step = 4
+    result = do_sinusoid_regression(X[::step], Y[::step], (1, 1, 1, 1))
```

Comment:
What's the point of `step` in these tests?

Reply:
You know I'm not sure. Just testing regression capabilities on a subset of data.

Reply:
good point!! fixed.